trpcultivate_phenotypes/src/Plugin/TripalImporter/TripalCultivatePhenotypesTraitsImporter.php

```diff
@@ -483,84 +401,208 @@ public function formValidate($form, &$form_state) {
+   */
+  public function processValidationMessages($failures) {
+    // Array to hold all validation results for each validator. Everything is
+    // set to status of 'todo' to start and will only change to one of 'pass' or
+    // 'fail' if the $failures[] array is defined for that validator, indicating
+    // that validation did take place.
+    $messages = [
+      // ----------------------------- METADATA --------------------------------
+      'GENUS' => [
+        'title' => 'Genus exists and/or matches the project/experiment',
+        'status' => 'todo',
+        'details' => ''
+      ],
+      // ------------------------------- FILE ----------------------------------
+      'FILE' => [
+        'title' => 'File is a valid tsv or txt',
+        'status' => 'todo',
+        'details' => ''
+      ],
+      // ---------------------------- HEADER ROW -------------------------------
+      'HEADERS' => [
+        'title' => 'File has all of the column headers expected',
+        'status' => 'todo',
+        'details' => ''
+      ],
+      // ----------------------------- DATA ROW --------------------------------
+      'empty_cell' => [
+        'title' => 'Required cells contain a value',
+        'status' => 'todo',
+        'details' => ''
+      ],
+      'valid_data_type' => [
+        'title' => 'Values in required cells are valid',
+        'status' => 'todo',
+        'details' => ''
+      ],
+      'duplicate_traits' => [
+        'title' => 'All trait-method-unit combinations are unique',
+        'status' => 'todo',
+        'details' => ''
+      ]
+    ];
+
+    foreach($messages as $validator_name => $default_messages) {
+      // Check if this validator exists in the failures array, which indicates
+      // that it was run.
+      if (array_key_exists($validator_name, $failures)) {
+
+        // ----------------------------- PASS ----------------------------------
+        // Check if $failures[$validator_name] is empty, which indicates there
+        // are no errors to report for this validator.
+        if (count($failures[$validator_name]) === 0 ) {
+          $messages[$validator_name] = [
+            'status' => 'pass',
+          ];
+        }
+
+        // ----------------------------- FAIL ----------------------------------
+        // Check if $failures[$validator_name] contains one of the results
+        // keys, indicating that this is not a row-level validator and therefore
+        // doesn't keep track of line numbers.
+        else if (array_key_exists('case', $failures[$validator_name])) {
+          // @todo: Update the message to not use the 'case' string by default
+          // and to incorporate the 'failed_details'.
+          $message = $failures[$validator_name]['case'];
+          $messages[$validator_name] = [
+            'status' => 'fail',
+            'details' => $message
+          ];
+          $messages['raw_results'] = $failures[$validator_name];
```

Comment:
For these we want to ensure the raw results are at the same level of the array as the status/details. So it should be this instead:
```suggestion
            'status' => 'fail',
            'details' => $message,
            'raw_results' => $failures[$validator_name],
          ];
```